node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+|Payable Account                            Balance Wei
+|
+|                                           Original
+|                                           Adjusted
+|
+|0x0000000000000000000000000000000000676869 {}
+|                                           {}
+|0x0000000000000000000000000000000000646566 {}
+|                                           {}
+|0x0000000000000000000000000000000000616263 {}
+|                                           {}",
+            balance_3.separate_with_commas(),
+            expected_adjusted_balance_3.separate_with_commas(),
+            balance_2.separate_with_commas(),
+            expected_adjusted_balance_2.separate_with_commas(),
+            balance_1.separate_with_commas(),
+            expected_adjusted_balance_1.separate_with_commas()
+        );
+        TestLogHandler::new().exists_log_containing(&log_msg.replace("|", ""));
+        test_inner_was_reset_to_null(subject)
+    }
+
+    #[test]
+    fn only_transaction_fee_causes_limitations_and_the_service_fee_balance_suffices() {
+        init_test_logging();
+        let test_name =
+            "only_transaction_fee_causes_limitations_and_the_service_fee_balance_suffices";
+        let now = SystemTime::now();
+        let balance_1 = multiple_by_billion(111_000_000);
+        let account_1 = make_plucked_qualified_account("abc", balance_1, 100_000_000, 20_000_000);
+        let balance_2 = multiple_by_billion(300_000_000);
+        let account_2 = make_plucked_qualified_account("def", balance_2, 120_000_000, 50_000_000);
+        let balance_3 = multiple_by_billion(222_222_222);
+        let account_3 = make_plucked_qualified_account("ghi", balance_3, 100_000_000, 40_000_000);
+        let qualified_payables = vec![account_1.clone(), account_2, account_3.clone()];
+        let analyzed_payables = convert_collection(qualified_payables);
+        let calculator_mock = CriterionCalculatorMock::default()
+            .calculate_result(multiple_by_billion(400_000_000))
+            // This account will be cut off because it has the lowest weight and only two accounts
+            // can be kept according to the limitations detected in the transaction fee balance
+            .calculate_result(multiple_by_billion(120_000_000))
+            .calculate_result(multiple_by_billion(250_000_000));
+        let mut subject = PaymentAdjusterReal::new();
+        subject.calculators = vec![Box::new(calculator_mock)];
+        subject.logger = Logger::new(test_name);
+        let agent_id_stamp = ArbitraryIdStamp::new();
+        let agent = BlockchainAgentMock::default()
+            .set_arbitrary_id_stamp(agent_id_stamp)
+            .service_fee_balance_minor_result(u128::MAX);
+        let adjustment_setup = PreparedAdjustment {
+            agent: Box::new(agent),
+            adjustment_analysis: AdjustmentAnalysis::new(
+                Adjustment::TransactionFeeInPriority {
+                    affordable_transaction_count: 2,
```

Comment:
Let's create a variable for the count somewhere during the preparatory statements of the test.